File: src/main/java/net/clgd/ccemux/emulation/tror/CapabilitiesPacket.java
package net.clgd.ccemux.emulation.tror;

import java.util.Set;

public class CapabilitiesPacket extends TRoRPacket<Set<String>> {
	public CapabilitiesPacket(Set<String> capabilities) {
		data = capabilities;
	}
	
	@Override
	public String getPacketCode() {
		return "SP";
	}
	
	@Override
	public String toString(String metadata) {
		return "SP:" + metadata + ";-" + (data.stream().reduce((s1, s2) -> s2 + s1)) + "-\n";
	}
	
	@Override
	public String toString() {
		return toString("");
	}
}

File: src/main/kotlin/net/clgd/ccemux/emulation/tror/CapabilitiesPacket.kt
package net.clgd.ccemux.emulation.tror

class CapabilitiesPacket(val caps: Set<String>) : TRoRPacket<Set<String>>(caps) {
	override val packetCode: String = "SP"

	override fun toString(metadata: String) = "SP:%s;-%s-\n".format(metadata, contents.reduce { a, b -> a + '-' + b })

	override fun toString() = toString("")
}
